### PR TITLE
Remove client status from allocation TerminalStatus

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

File: nomad/core_sched.go [hunk 1]
```diff
@@ -198,7 +198,12 @@ func (c *CoreScheduler) gcEval(eval *structs.Evaluation, thresholdIndex uint64)
 
 	// Scan the allocations to ensure they are terminal and old
 	for _, alloc := range allocs {
-		if !alloc.TerminalStatus() || alloc.ModifyIndex > thresholdIndex {
+		// TODO: This can go away once the scheduler marks an alloc as
+		// DesiredStatusStop when a client fails.
```
Comment (diptanu) [inline nomad/core_sched.go]: Shouldn't it be DesiredStatusDead?

**Comment by diptanu** [comment]
LGTM, just had a small question.

File: nomad/core_sched.go (hunk 1 above)
Comment (dadgar) [inline nomad/core_sched.go]: The Client has state Dead/Complete. The server just sets desired state Stop

**Comment by diptanu** [comment]
@dadgar This PR has broken. Basically to check if a periodic job has running allocation we do the following -

```
	for _, alloc := range allocs {
				if !alloc.TerminalStatus() {
					return true, nil
				}
			}
```

This always returns true since we no longer check the client status.  And hence for periodic jobs where users don't want overlapping jobs the periodic jobs don't trigger any longer.

Fix for this is #1016

**Pull request merged**